benchexec/resources.py

```diff
@@ -15,191 +10,157 @@
 import math
 import os
 import sys
+from functools import cmp_to_key
 
-from benchexec import cgroups
-from benchexec import util
+import cgroups
+import util
 
-__all__ = [
-    "check_memory_size",
-    "get_cpu_cores_per_run",
-    "get_memory_banks_per_run",
-    "get_cpu_package_for_core",
-]
-
-
-def get_cpu_cores_per_run(
-    coreLimit, num_of_threads, use_hyperthreading, my_cgroups, coreSet=None
-):
-    """
-    Calculate an assignment of the available CPU cores to a number
-    of parallel benchmark executions such that each run gets its own cores
-    without overlapping of cores between runs.
-    In case the machine has hyper-threading, this method tries to avoid
-    putting two different runs on the same physical core
-    (but it does not guarantee this if the number of parallel runs is too high to avoid it).
-    In case the machine has multiple CPUs, this method avoids
-    splitting a run across multiple CPUs if the number of cores per run
-    is lower than the number of cores per CPU
-    (splitting a run over multiple CPUs provides worse performance).
-    It will also try to split the runs evenly across all available CPUs.
-
-    A few theoretically-possible cases are not implemented,
-    for example assigning three 10-core runs on a machine
-    with two 16-core CPUs (this would have unfair core assignment
-    and thus undesirable performance characteristics anyway).
-
-    The list of available cores is read from the cgroup file system,
-    such that the assigned cores are a subset of the cores
-    that the current process is allowed to use.
-    This script does currently not support situations
-    where the available cores are asymmetrically split over CPUs,
-    e.g. 3 cores on one CPU and 5 on another.
-
-    @param coreLimit: the number of cores for each run
-    @param num_of_threads: the number of parallel benchmark executions
-    @param coreSet: the list of CPU cores identifiers provided by a user, None makes benchexec using all cores
-    @return a list of lists, where each inner list contains the cores for one run
-    """
+# prepping function, consider change of name
+def get_cpu_cores_per_run(  
```

Comment:
Text explaining functions for callers should be doc strings (string at first line of the function) and not comments. This actually attaches the text with the explanation to the function in Python.

Please use this style for all functions, and extensively document them, their requirements, their guarantees, and their parameters and return values.

Reply:
I hope that all are done now ... let me know if you find a docstring missing.